recycleIT-frontend/src/Components/Services/Map/Map.tsx

```diff
@@ -1,29 +1,111 @@
 
-const AnyReactComponent = ({ text }:any) => <div>{text}</div>;
+const UserLocationMarker = (props: LocationType) => {
+  return (
+    <Tooltip 
+      arrow 
+      title="Me" 
+      placement="top" 
+      style={{backgroundColor: 'white', color: 'black'}}>
+        <PersonPinCircleIcon fontSize='large' style={{color: '#1034a6'}}/>
+    </Tooltip>
+  )
+}
 
-export default function Map(){
+export default function Map(props: IMapProps){
   const defaultProps = {
     center: {
-      lat: 10.99835602,
-      lng: 77.01502627
+      lat: 40.730610,
+      lng: -73.935242
     },
-    zoom: 11
+    zoom: 5
   };
+  
+  const chosenItemLocation = props.chosenItemLocation as ChosenItemType;
+  const userLocation = useContext(UserLocation) as LocationType;
+  const [ center, setCenter ] = useState(defaultProps.center);
+  const [ zoom, setZoom ] = useState(defaultProps.zoom);
+  const [ chosenItemId, setChosenItemId ] = useState(0);
+  
+  useEffect(() => {
+    setCenter(chosenItemLocation);
+    setZoom(10);
+    if (chosenItemLocation) {
+      setChosenItemId(chosenItemLocation.id);
+    }
+  }, [chosenItemLocation])
+
+  useEffect(() => {
+    setCenter(userLocation);
+  }, [userLocation]);
+
+  useEffect(() => {
+    if (props.servicesList) {
+      setCenter({
+        lat: props.servicesList[0].latitude, 
+        lng: props.servicesList[0].longitude
+      });
+      setZoom(8);
+    }
+  }, [props.servicesList])
+
+  let markers = null;
+
+  if (props.servicesList) {
+    markers = props.servicesList.map(service => (
+      <Marker 
+          lat={service.latitude} 
+          lng={service.longitude}
+          name={service.name}
+          rating={service.raiting}
+          key={service.id}
+          id={service.id}
+          showLabel={chosenItemId}
+      />
+    ))
+  }
 
   return (
-    <div style={{ height: '700px', width: '70%', marginLeft: '450px' }}>
-      <GoogleMapReact
-        bootstrapURLKeys={{ key: "AIzaSyBX9wy2s_ljVzoNiOBc_ki0fAFpgJCj2G4" }}
-        defaultCenter={defaultProps.center}
-        defaultZoom={defaultProps.zoom}
-      >
-        <AnyReactComponent
-          lat={59.955413}
-          lng={30.337844}
-          text="My Marker"
-        />
-      </GoogleMapReact>
+    <div className="map-side">
+      <Typography gutterBottom variant="h6" style={{color: '#4F4F4F'}}>
+        <IconButton 
+          style={{display: !props.isMobileDevice ? 'inline-block' : 'none'}} 
+          onClick={props.toggleDrawerOpened}>
+            <MenuOpenIcon style={{color: '#4F4F4F'}} />
+        </IconButton>
+          Search for eco services, available for you
+        </Typography>
+      <div className="map-container">
+        <GoogleMapReact
+          bootstrapURLKeys={{ key: "AIzaSyBX9wy2s_ljVzoNiOBc_ki0fAFpgJCj2G4" }}
+          defaultCenter={center}
+          center={center}
+          defaultZoom={zoom}
+          zoom={zoom}
+        >
+          <UserLocationMarker
```

Comment:
To adjust marker position check this sample code, it works fine when zooming
https://github.com/google-map-react/old-examples/tree/master/web/flux/components/examples/x_simple